### PR TITLE
Added decodePlist (decode file at url using PropertyListDecoder)

### PR DESCRIPTION
**Pull request by AlexeiGitH**
Decoder method to complete #603.

## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 4.
- [x] New extensions support iOS 8.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [x] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,8 @@ The changelog for **SwifterSwift**. Also see the [releases](https://github.com/S
 # Upcoming release
 
 ### Added
+- **FileManager**
+- Added `decodePlist(as:using:from:)` method to decode saved Encodable object from file URL using PropertyListDecoder. [#614](https://github.com/SwifterSwift/SwifterSwift/pull/614*) by [AlexeiGitH](https://github.com/AlexeiGitH)
```
Comment (guykogus) [inline CHANGELOG.md]: The object doesn't have to conform to `Encodable` to be decoded.
```suggestion
- Added `decodePlist(as:using:from:)` method to decode an object from file using `PropertyListDecoder`. [#614](https://github.com/SwifterSwift/SwifterSwift/pull/614) by [AlexeiGitH](https://github.com/AlexeiGitH)
```

File: Sources/Extensions/Foundation/FileManagerExtensions.swift [hunk 2]
```diff
@@ -56,5 +56,19 @@ public extension FileManager {
...
+    /// SwifterSwift: Decode Encodable object from file using PropertyListDecoder.
+    ///
+    /// - Parameters:
+    ///   - type: type of the object that is being decoded.
+    ///   - decoder: PropertyListDecoder.
+    ///   - url: URL to a file  with saved object data.
+    /// - Throws: Throws any errors thrown by JSONDecoder.
+    /// - Returns: An object of specified type. Returns nil if failed to read from URL.
+    public func decodePlist<T: Decodable>(as type: T.Type, using decoder: PropertyListDecoder = PropertyListDecoder(), from url: URL) throws -> T? {
+        guard let data = FileManager.default.contents(atPath: url.path) else {
```
Comment (guykogus) [inline Sources/Extensions/Foundation/FileManagerExtensions.swift]: You can put the `return` on the same line to save space.
```suggestion
        guard let data = FileManager.default.contents(atPath: url.path) else { return nil }
```

File: Tests/FoundationTests/FileManagerExtensionsTests.swift [hunk 3]
```diff
@@ -80,6 +80,30 @@ final class FileManagerExtensionsTests: XCTestCase {
...
+            
+            XCTAssert(FileManager.default.fileExists(atPath: fileURL.path))
+            
+            let fileContents = FileManager.default.contents(atPath: fileURL.path)
+            XCTAssertNotNil(fileContents)
+            
+            let decoder = PropertyListDecoder()
+            let decodedPoint = try decoder.decode(CGPoint.self, from: fileContents!)
+            
+            XCTAssert(decodedPoint == point)
```
Comment (guykogus) [inline Tests/FoundationTests/FileManagerExtensionsTests.swift]: Use `XCTAssertEqual`
```suggestion
            XCTAssertEqual(decodedPoint, point)
```

File: Sources/Extensions/Foundation/FileManagerExtensions.swift [hunk 4]
```diff
@@ -56,5 +56,19 @@ public extension FileManager {
         return nil
     }
 
+    /// SwifterSwift: Decode Encodable object from file using PropertyListDecoder.
```
Comment (guykogus) [inline Sources/Extensions/Foundation/FileManagerExtensions.swift]:
```suggestion
    /// SwifterSwift: Decode an object from file using PropertyListDecoder.
```

File: Sources/Extensions/Foundation/FileManagerExtensions.swift [hunk 5]
```diff
@@ -56,5 +56,19 @@ public extension FileManager {
...
+    ///
+    /// - Parameters:
+    ///   - type: type of the object that is being decoded.
+    ///   - decoder: PropertyListDecoder.
+    ///   - url: URL to a file  with saved object data.
+    /// - Throws: Throws any errors thrown by JSONDecoder.
```
Comment (guykogus) [inline Sources/Extensions/Foundation/FileManagerExtensions.swift]:
```suggestion
    /// - Throws: Throws any errors thrown by `PropertyListDecoder`.
```

**Comment by LucianoPAlmeida** [comment]
Hey @AlexeiGitH Can you just resolve the conflicts with master? After that I think it will be good to go :)

File: Tests/FoundationTests/FileManagerExtensionsTests.swift [hunk 6]
```diff
@@ -80,6 +80,31 @@ final class FileManagerExtensionsTests: XCTestCase {
...
+        let temporaryFilename = ProcessInfo().globallyUniqueString
+        let fileURL = temporaryDirectoryURL.appendingPathComponent(temporaryFilename)
+        do {
+            let encoder = PropertyListEncoder()
+            let dataToEncode = try encoder.encode(point)
+            FileManager.default.createFile(atPath: fileURL.path, contents: dataToEncode, attributes: nil)
+            
+            XCTAssert(FileManager.default.fileExists(atPath: fileURL.path))
+            
+            let fileContents = FileManager.default.contents(atPath: fileURL.path)
```
Comment (guykogus) [inline Tests/FoundationTests/FileManagerExtensionsTests.swift]: You're actually meant to be using the `decodePlist(as:using:)` function you created in this test 🤣

**Comment by guykogus** [comment]
@omaralbeik @LucianoPAlmeida can you please check?

**Comment by LucianoPAlmeida** [comment]
Just cleaning up those old PR's. 
@AlexeiGitH Closing this one due to inactivity, feel free to reopen :))

**Pull request closed without merging**